module/zfs/zio.c

```diff
@@ -4153,6 +4153,8 @@ zio_encrypt(zio_t *zio)
 	ASSERT3U(psize, !=, 0);
 
 	enc_buf = zio_buf_alloc(psize);
+	bzero(enc_buf, psize);
+
 	eabd = abd_get_from_buf(enc_buf, psize);
 	abd_take_ownership_of_buf(eabd, B_TRUE);
```

Comment:
This isn't really related to your change, but I'm wondering if aside from the zeroing, is this equivalent to the simpler:
```
eabd = abd_alloc_linear(psize);
enc_buf = abd_to_buf(eabd);
```
?

Reply:
Yes, that looks to me like it would be the same.  If you like I could rework it.